components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -53,27 +55,39 @@
     private static final Log log = LogFactory.getLog(UserEmailVerificationHandler.class);
 
     public String getName() {
+
         return "userEmailVerification";
     }
 
     public String getFriendlyName() {
+
         return "User Email Verification";
     }
 
     @Override
     public void handleEvent(Event event) throws IdentityEventException {
 
         Map<String, Object> eventProperties = event.getEventProperties();
+        String eventName = event.getEventName();
         UserStoreManager userStoreManager = (UserStoreManager) eventProperties.get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
         User user = getUser(eventProperties, userStoreManager);
 
-        boolean enable = Boolean.parseBoolean(Utils.getConnectorConfig(
-                IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMIL_VERIFICATION, user.getTenantDomain()));
+        boolean enable = false;
+        if (IdentityEventConstants.Event.PRE_ADD_USER.equals(eventName) ||
+                IdentityEventConstants.Event.POST_ADD_USER.equals(eventName)) {
+            enable = Boolean.parseBoolean(Utils.getConnectorConfig(IdentityRecoveryConstants.ConnectorConfig
+                    .ENABLE_EMIL_VERIFICATION, user.getTenantDomain()));
+        } else if (IdentityEventConstants.Event.PRE_SET_USER_CLAIMS.equals(eventName) ||
+                IdentityEventConstants.Event.POST_SET_USER_CLAIMS.equals(eventName)) {
+            enable = Boolean.parseBoolean(Utils.getConnectorConfig(IdentityRecoveryConstants.ConnectorConfig
+                    .ENABLE_EMAIL_VERIFICATION_ON_UPDATE, user.getTenantDomain()));
+        }
 
         if (!enable) {
-            //Email Verification feature is disabled
+            // Email Verification feature is disabled.
             if (log.isDebugEnabled()) {
-                log.debug("Email verification Handler is disabled in tenant: " + user.getTenantDomain());
+                log.debug("Email verification Handler is disabled in tenant: " + user.getTenantDomain() + "for " +
+                        "adding user or updating user claims.");
```

Comment:
Shall we log the event name and the handler status

Reply:
tenant: <tenant> for event : <event>

Reply:
changed the debug log accordingly.